src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -778,6 +778,15 @@ abstract class BTypes {
       } while (fcs == null)
       fcs
     }
+
+    // equallity and hashcode is based on internalName
+    override def equals(obj: scala.Any): Boolean = obj match {
```

Comment:
I think reference equality / identity hashCode should be fine here, no? The construction pattern should guarantee that there's a single instance per `InternalName`.

Reply:
i did this because it failed some unit tests. It is nt used in production though,just a test that ran from a symbol and from a class file